### PR TITLE
Release candidate 4.34

### PR DESCRIPTION
**Pull request by dnil**
This PR marks a new Scout release. We apply semantic versioning. This is a minor release since several errors were corrected, some UI functionality was added, but the core API is unchanged.

**How to test**:
### Added
- [x] Saved filter lock and unlock
- [x] Filters can optionally be marked audited, logging the filter name, user and date on the case events and general report.
- [x] Added `ClinVar hits` and `Cosmic hits` in cancer SNVs filters
- [x] Added `ClinVar hits` to variants filter (rare disease track)
- [x] Load cancer demo case in docker-compose files (default and demo file)
- [x] Inclusive-language check using [woke](https://github.com/get-woke/woke) github action
- [x] Add link to HmtVar for mitochondrial variants (if VCF is annotated with HmtNote)
- [x] Grey background for dismissed compounds in variants list and variant page
- [x] Pin badge for pinned compounds in variants list and variant page
- [x] Support LoqusDB REST API queries
- [x] Add a docker-compose-matchmaker under scout/containers/development to test matchmaker locally
- [x] Script to investigate consequences of symbol search bug
~~- [ ] Added GATK to list of SV and cancer SV callers~~ reported ok
### Fixed
~~- [ ] Make MitoMap link work for hg38 again~~ reported ok
- [x] Export Variants feature crashing when one of the variants has no primary transcripts
- [x] Redirect to last visited variantS page when dismissing variants from variants list
- [x] Improved matching of SVs Loqus occurrences in other cases
- [x] Remove padding from the list inside (Matching causatives from other cases) panel
- [x] Pass None to get_app function in CLI base since passing script_info to app factory functions was deprecated in Flask 2.0
- [x] Fixed failing tests due to Flask update to version 2.0
- [x] Speed up user events view
- [ ] Causative view sort out of memory error
- [x] Use hgnc_id for gene filter query
- [x] Typo in case controllers displaying an error every time a patient is matched against external MatchMaker nodes
- [x] Do not crash while attemping an update for variant documents that are too big (> 16 MB)
- [x] Old STR causatives (and other variants) may not have HGNC symbols - fix sort lambda
- [x] Check if gene_obj has primary_transcript before trying to access it
- [x] Warn if a gene manually searched is in a clinical panel with an outdated name when filtering variants
### Changed
- [x] Remove parsing of case `genome_version`, since it's not used anywhere downstream
- [x] Introduce deprecation warning for Loqus configs that are not dictionaries
- [x] SV clinical filter no longer filters out sub 100 nt variants
- [x] Count cases in LoqusDB by variant type
- [x] Commit pulse repo badge temporarily set to weekly
- [x] Sort ClinVar submissions objects by ascending "Last evaluated" date
- [x] Refactored the MatchMaker integration as an extension
- [x] Replaced some sensitive words as suggested by woke linter
- [x] Documentation for load-configuration rewritten.
- [x] Add styles to MatchMaker matches table
- [x] More detailed info on the data shared in MatchMaker submission form

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: docs/blog/new-4.34.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+## What's new in 4.34?
+
+_Posted: May 17 2021_
+
+This is a relatively large release, forced out by the need to fix a **problem affecting gene panels with outdated gene symbols starting in v4.21**.
+
+### Gene panel HGNC symbol search bug
+The 4.21 update allowed freer use of gene panels as lists, as well as the application of the latest version of the panel to each case.
+HGNC symbols were previously handled by a separate mechanism, and subject to some validation checks.
+Regrettably, HGNC symbols were taken as correct from the panels. This lead to a situation where if the gene symbol had been updated in HGNC
```
Comment (northwestwitch) [inline docs/blog/new-4.34.md:10]:
```suggestion
Regrettably, HGNC symbols were taken as correct from the panels. This led to a situation where if the gene symbol had been updated in HGNC
```

File: docs/blog/new-4.34.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+but not on the latest panel version, variants in that gene could be missed on panel search. This fixed in this release, and we now always use the HGNC id
```
Comment (northwestwitch) [inline docs/blog/new-4.34.md:11]:
```suggestion
but not on the latest panel version, variants in that gene could be missed on panel search. This is fixed in this release, and we now always use the HGNC id
```

File: docs/blog/new-4.34.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+behind the scenes for panel searches as well as manually entered HGNC symbol searches.
+
+If you are on another instance than Solna, and want to check your database for panels, genes, cases and variants that could have been so compromised,
+please find a new script in the scripts directory that you can modify to reflect your deploy date of 4.21.
+
+### Other features
```
Comment (northwestwitch) [inline docs/blog/new-4.34.md:17]: Fantastic! 🥇

**Review by northwestwitch — APPROVED** [review]
Great job with the new blog post! 👍🏻

**Comment by northwestwitch** [comment]
This branch is installed on Clinical-db stage now!

**Comment by northwestwitch** [comment]
Nooo I've pushed to the wrong branch!! I'll fix, sorry!

**Comment by northwestwitch** [comment]
Checking the item `Improved matching of SVs Loqus occurrences in other cases` with another [variant](https://scout-stage.scilifelab.se/cust000/643594-300M/sv/variants/4f0ac402eec5dcbe854649b3726f0d96) that has 25 matches but shows none (at least to me):

Loqus command: 
```/home/hiseq.clinical/miniconda/envs/S_loqusdb/bin/loqusdb --config /home/hiseq.clinical/servers/config/clinical-db.scilifelab.se/loqusdb.yaml variants --to-json --variant-id 7_74155001_C_DUP --case-count -t sv -c 7 -s 74155001 -e 74195000  --end-chromosome 7 --sv-type DUP```

Returns:
`{"chrom": "7", "end_chrom": "7", "sv_type": "DUP", "pos_sum": 1853879025, "end_sum": 1854876000, "observations": 25, "length": 39879, "families": ["givingleech", "cleardassie", "tidywhale", "livingtoad", "propercod", "quietmarmot", "lightpig", "awarepuma", "meetlark", "intentbream", "provenswine", "actualinsect", "cosmicwalrus", "savinglynx", "busyprawn", "kindpika", "vastcougar", "calmlemur", "validsnipe", "manydane", "nicesponge", "readymagpie", "neatshrimp", "smartfowl", "soundclam"], "pos_left": 74153161, "pos_right": 74157161, "end_left": 74193040, "end_right": 74197040, "total": 4503`

givingleech --> cust002, only on prod
cleardassie --> cust002, only on prod
tidywhale    --> cust003, only on prod
livingtoad   --> cust002, only on prod
propercod  --> cust002, only on prod
quietmarmot --> cust002, only on prod
lightpig --> cust003, only on prod
awarepuma --> cust002, only on prod
meetlark --> cust002, only on prod
intentbream --> cust003, only on prod
provenswine --> cust002, only on prod

I could continue checking all 25 but I think we are good, also considering that we tested quite well the PR?

**Comment by dnil** [comment]
I agree - I have also been checking around for a bit now but have so far not found any errors with the similar-SV part. I think we just have to go with what we have there, see it live and take any deltas then.

**Comment by northwestwitch** [comment]
What about the `Causative view sort out of memory error`?

**Comment by dnil** [comment]
> What about the `Causative view sort out of memory error`?

It's about the same there: it is working on stage, which in all honesty 4.33 also did, so just fingers crossed it will all be ok on prod. It is already broken, so should not be worse.

Mental note for deploying also to do the index update. I suggest that we do that manually so as not to inadvertently trigger any big rebuilds with `scout update index`. Agreed?

**Comment by northwestwitch** [comment]
> Agreed?

Yes!

**Pull request merged**